enclave_manager/avalon_enclave_manager/kme_listener.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import logging
+import argparse
+
+from jsonrpc.dispatcher import Dispatcher
+
+import config.config as pconfig
+import utility.logger as plogger
+from listener.base_jrpc_listener \
+    import BaseJRPCListener, start_listener, parse_bind_url, get_config_dir
+
+logger = logging.getLogger(__name__)
+
+
+class KMEListener(BaseJRPCListener):
+    """
+    Listener to handle requests from WorkerProcessingEnclave(WPE)
+    """
+
+    # The isLeaf instance variable describes whether a resource will have
+    # children and only leaf resources get rendered. KMEListener is a leaf
+    # node in the derivation tree and hence isLeaf is required.
+    isLeaf = True
+
+    def __init__(self, config):
+        self.config = config
+        self.dispatcher = Dispatcher()
+        self.kme_request_handler = None
+
+        # TODO : Uncomment following statements after
+        # a request handler is created
+        '''self.dispatcher.add_method(self.kme_request_handler\
+            .RegisterWorkorderProcessor)
+        self.dispatcher.add_method(self.kme_request_handler\
+            .PreporocessWorkorder)'''
```

Comment:
`config`, `dispatcher` could be used in base class. Also create `Dispatcher()` in base class and call `dispatcher.add_method()` in base class . Pass the array with rpc method names to base class constructor and add the methods in dispatcher in a loop.

```
class BaseJRPCListener(resource.Resource):
    def __init__(self, config, rpc_methods):
        self.config = config
        self.dispatcher = Dispatcher()
        for m in rpc_methods:
            self.dispatcher.add_method(m)

class KMEListener(BaseJRPCListener):
    def __init__(self, config):
        rpc_methods = [ RegisterWorkorderProcessor,                                   
                                    PreporocessWorkorder]
        super().__init__(config, rpc_methods)      
```